Explicitly set magit-section-highlight to grey20 background

ska2342 opened the pull request:
Magit uses magit-section-highlight for the current section. That was unreadable.
This commits makes that face use just a black background.

Commit:
Add magit-section-highlight with black background

jasonm23 commented:
Please add a screen shot to this conversation thread.

ska2342 commented:
Before:

![screenshot_20170310_092003](https://cloud.githubusercontent.com/assets/972373/23787568/facce6c8-0572-11e7-82d0-85e89203c0b6.png)

After:

![screenshot_20170310_092109](https://cloud.githubusercontent.com/assets/972373/23787580/04b0ae90-0573-11e7-80a9-d1005ff96b98.png)


I am on Emacs 24.5.1 with magit 20170308.1519 and darktooth-theme Package-Version: 20170216.2034, Version: 0.3.4

dieggsy commented:
Is this a new change in magit? I can't recall ever having this issue.

**EDIT:** Updated everything, this is what magit looks like for me:
![2017-03-10_09 29 09](https://cloud.githubusercontent.com/assets/9020453/23798785/1b732094-0574-11e7-87c8-ddc24f71b210.png)

On emacs 26.0.50 and magit 20170308.1519, with my only slightly modified darktooth theme (but importantly I made no changes to the magit-section-highlight face).

ska2342 commented:
Hm, according to https://github.com/magit/magit/blame/master/lisp/magit-section.el#L110 that face has been around for years.

I can see that other themes set it:
* https://github.com/bbatsov/zenburn-emacs/blob/master/zenburn-theme.el#L816
* https://github.com/NicolasPetton/zerodark-theme/blob/master/zerodark-theme.el#L376

If you want me to, I can try to create a new user, install magit and darktooth only and see what's up. Given my amount of customizations, it is quite possible that something interferes. Unfortunately, emacs -q is not really helpful for testing this, as I need to install at least magit to test.

jasonm23 commented:
@ska2342 please try the current version of the theme with a current version of Emacs stable (25.1)

Happy to merge if this patch looks ok on stable also.

dieggsy commented:
@jasonm23 i'm not sure what the cause of it being illegible for @ska2342 is, but i think the way it looks in my screenshot (eg making the background slightly lighter) is slightly more visually appealing that the greater contrast of making the background darker.

By the way @ska2342, for me the face is set to `grey20`, which is what it's set to in the magit source code. At least in my Emacs (though I doubt this depends on version that greatly) this color is darker than your first screenshot.

ska2342 commented:
Hi,

I am also baffled, why it doesn't work out of the box for me.  Tried a few things, to hunt this down:

* New user with Unity: all is well
* New user with Plasma: all is well
* New user with Plasma, breeze dark theme: all is well
* My user in MELPA sandbox: all is well
* My user with emacs -Q, added MELPA, switched to darktooth: all is well
* My user with emacs -q, added MELPA, switched to darktooth: all is well
* My user with my init.el moved away, added MELPA, switched to darktooth: all is well
* My user with just my custom.el moved away. Broken

So, it is due to something in my init.el.  Since I don't mess with faces there, I am now totally clueless what is causing this.

Thus, I moved my init.el out of the way and evaluated it form by form, always keeping an eye on an open magit status buffer. The form that finally triggered the bad display was

```elisp
(when (load-theme 'darktooth)
  (enable-theme 'darktooth)
  (darktooth-modeline))
```
When I remove those lines, everything works fine. I added them to allow turning on the modeline. The theme is already activated by customize which is loaded earlier.

If you have any clue what could be causing this, I'd be happy to hear. How am I supposed to activate the modeline if not by calling it like this?

As other themes set this font explicitly, this PR (even with a different font) could still be useful. But I am not really familiar with the theme engine.

Hope that helps.

jasonm23 commented:
That's interesting behavior, and I'm no more able to explain it than you.

Did you try all these things with 25.1 or your existing Emacs 24?

ska2342 commented:
I did all this on my existing Emacs 24. I wanted to avoid the effort of installing a new Emacs for a 1-line patch which, looking back, is kinda funny.

And it got even weirder: When I keep in init.el in place, comment it out, comment it form-by-form back in, and restart Emacs after every edit, it breaks at a different place: when I inhibit the startup screen.

There something very fishy going on.

Still, with this patch everything works under all conditions on my machine.

ska2342 commented:
Compiled my own 25.1 (installed build-deps for emacs24, fetched tar, configure/make, run emacs from src subdir).

* I can reproduce my weird problem.
* I can confirm that this patch fixes it.

ska2342 commented:
When I run `autothemer-generate-templates` I see two variants of `magit-section-highlight`:

When everything looks good, I see 

```elisp
 (magit-section-highlight
  (:background darktooth-dark0_soft))
```

In the case of the bad looking display, I get 

```elisp
 (magit-section-highlight
  (:background darktooth-white))
```
To me, this looks like the face is somehow derived from somewhere else but depending on whatever is causing this weirdness it can be set to different values.

Setting it explicitly makes things stable, at least for me.

Commit:
Change magit-section-highlight to default.

By default, magit-section-highlight takes on the value
darktooth-dark0_soft. On some systems, when not set explicitly,
magit-section-highlight computes to a different background.

Thus, set the background explicitly to the common default
value. Replacing the former black background.

ska2342 commented:
I changed this PR to set `magit-section-highlight` to `darktooth-dark0_soft` which seems to be the common default.

jasonm23 commented:
Thanks for checking with 25.1 

Merged

ska2342 commented:
Thanks for accepting. What a journey for such a small change! :-)
